subscriptions-api.md

```diff
@@ -0,0 +1,613 @@
+
+#### Intermediary
+
+An "intermediary" (also referred to as middleware) receives an event for the
+purpose of forwarding it to the next receiver, which might be another
+intermediary or a consumer. A typical task for an intermediary is to route the
+event to receivers based on the information in the event context.
+
+#### Consumer
+
+An entity receiving and processing events. Consumers may receive events from
+producers directly or via intermediaries. A consumer might listen and wait for
+events to be delivered to it or it might actively solicit them from the producer
+or intermediary.
+
+#### Subscription
+
+A relationship established between a consumer and a producer or an intermediary.
+The subscription reflects the consumer's interest in receiving events and
+describes the method for how to deliver those events.
+
+#### Subscription Manager
+
+An entity, defined in this specification, that manages the lifecycle of a
+subscription on behalf of an event consumer and that distributes events to
+registered consumers.
+
+## 3. Event Subscriptions
+
+This specification defines mechanisms for CNCF CloudEvents event consumers to
+subscribe to events originating from producers on behalf of event sources. The
+software entity handling these subscriptions and responsible for distributing
+events is abstractly referred to as a "subscription manager".
+
+A compliant CloudEvents subscription manager MUST support at least one of the
+transport protocols referenced in this specification and it MUST implement the
+subscription mechanism referred to or defined in this specification.
+
+In some cases, the event producer MAY delegate the role of the subscription
+manager to some intermediary, in which case produced events are made available
+to the intermediary, and the intermediary's subscription manager determines who
+receives copies of the published events. In other cases, the event producer MAY
+take on the subscription manager role by itself.
+
+In this specification, we distinguish two styles of relationships between an
+event consumer and a subscription manager, whereby the key differentiator is how
+the event delivery channel to the consumer is initiated:
+
+- Subscriptions with consumer-solicited delivery ("pull"-style) are configured
+  on the subscription manager for delivering events through a communication
+  channel (like an AMQP link or a MQTT connection) initiated by the consumer
+  and connecting to the subscription manager. These kinds of subscriptions are
+  typically offered by messaging or eventing middleware and their lifetime
+  might be bounded by the lifetime of the communication channel.
+
+- Subscriptions with subscription-manager-initiated delivery ("push"-style) are
+  configured on the subscription manager for delivering events through a
+  communication channel that the subscription manager initiates when events
+  are available for delivery on the subscription. The configuration of such a
```

Comment:
IMHO the accessory "when events are available for delivery on the subscription", although vague, could be misunderstood as if the communication channel has to be initiated at the time the subscription manager got an event. But the channel could have been setup before an event becomes available or the subscription manager could employ batching and delay the setup. Maybe the temporal clause, although illustrative, is not necessary here?

Reply:
Batching and delay or connection pooling are all optimizations, IMO. The subscription manager needs to initiate the channel when there are events to deliver. That could also just be once day, but if there are no events to deliver at all, it is likely not bothering.